src/WebJobs.Script.WebHost/WebHostServiceCollectionExtensions.cs

```diff
@@ -121,6 +122,7 @@ public static void AddWebJobsScriptHost(this IServiceCollection services, IConfi
             // Core script host services
             services.AddSingleton<WebJobsScriptHostService>();
             services.AddSingleton<IHostedService>(s => s.GetRequiredService<WebJobsScriptHostService>());
+            services.AddSingleton<IHostedService, HostedServiceManager>();
```

Comment:
I haven't been able to find resources on shutdown order, but is the assumption that services registered through `AddCommonServices` like `RpcInitializationService` are stopped by `StopAsync` before our `HostedServiceManager` shuts down things that are `IManagedHostedService`? If so, can we add a comment for that? 

I'm mainly not sure why `await ShutdownRpcServerAsync();` isn't called in `StopAsync` in `RpcInitializationService`

Reply:
Yes, having a comment in the interface method would be helpful.

Withe the web host builder, the order differs from the generic host builders and stop is called in the order services are registered (as opposed to the reverse order, which is the behavior of the generic host).

Reply:
I'm also adding a comment to the implementation, as ideally, the manager would reverse the order.

Reply:
If ShutdownRpcServerAsync() is called within StopAsync() of RpcInitializationService, it would always timeout and call the KillAsync() since _rpcServer.Shutdown() waits for LanguageWorkers to close before shutting itself down (You can try to change _rpcServerKill() to Shutdown() and do a ctrl+c, you'll notice that it'll just wait). 

LanguageWorker disposal is not initiated until RpcInitializationService's StopAsync() is complete (which would sunsequently initiate the StopAsync() of WebJobsScriptHostService which is what will eventually trigger the disposal of language workers). Therefore, introducing the ShutdownAsync() within StopAsync() would cause a deadlock.

Reply:
I also added comments both here and in the interface.